src/librustc_mir/build/expr/stmt.rs

```diff
@@ -161,6 +162,41 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                 );
                 block.unit()
             }
+            ExprKind::PlaceTypeAscription { source, user_ty } => {
```

Comment:
This code is not in the right place, I don't think. It should go into `expr_as_place`:

https://github.com/rust-lang/rust/blob/1002e404e1ae6b8b907c8655edd41380d0c561cb/src/librustc_mir/build/expr/as_place.rs#L45-L50

The idea is that each expr kind is implemented in exactly one place, whereever is the "best fit". This particular spot is for "statements" -- control-flow like things.